test/printf.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
```

Comment:
Oh, I guess it's obvious, the bug was it wrote too many bytes. asm2wasm layout probably happens to have nothing important after those bytes, while wasm-backend does.